### PR TITLE
Adding Hash Solidity compatible example

### LINKED ISSUE #14: feat: Solidity-compatible keccak hashes
In this example, we will show the example of a contract that uses keccak to hash the same values as Solidity.
Waiting for https://github.com/starkware-libs/cairo/issues/3477

### PR DESCRIPTION
**Pull request by julienbrs**
**Issue:**closes #14 

### Description

Added an example to get the same hash result as the solidity's keccak.

### CODE REVIEW

**Comment by julienbrs** [comment]
WIll add a contract address later, after the merge (so I can use remix to deploy easily)

File: src/ch02/hash-solidity-compatible.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+This contract demonstrates Keccak hashing in Cairo to match Solidity's keccak256. While both use Keccak, their endianness differs: Cairo is little-endian, Solidity big-endian. The contract achieves compatibility by hashing in big-endian using `keccak_u256s_be_inputs`, and reversing the bytes of the result with `u128_byte_reverse`.
+
+For example:
+
+```rust
+{{#include ../../listings/ch02-advanced-concepts/hash_solidity_compatible/src/contract.cairo}}
+```
+
+Visit contract on [Voyager](#) or play with it in [Remix](https://remix.ethereum.org/?#activate=Starknet&url=https://github.com/NethermindEth/StarknetByExample/blob/main/listings/ch02-advanced-concepts/hash_solidity_compatible/src/contract.cairo).
```
Comment (julio4) [inline src/ch02/hash-solidity-compatible.md:11]: If there's no deployed instance of the contract you can remove the "Visit contract on Voyager"

File: listings/ch02-advanced-concepts/hash_solidity_compatible/src/tests.cairo [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        let mut state = setup();
+        let mut array: Array<u256> = ArrayTrait::new();
+        array.append(1);
+
+        let hashExpected: u256 = 0xb10e2d527612073b26eecdfd717e6a320cf44b4afac2b0732d9fcbe2b7fa0cf6;
+
+        let hash = state.hash_data(array.span());
+        let hashReceived: u256 = state.get_hashed_value();
+
+        assert(hashReceived == hashExpected, 'hashReceived != hashExpected');
```
Comment (julio4) [inline listings/ch02-advanced-concepts/hash_solidity_compatible/src/tests.cairo:29]: use `snake_case` for variable names.

File: listings/ch02-advanced-concepts/hash_solidity_compatible/src/contract.cairo [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+            // Reverse merge the reversed segments back into a u256 value
+            let compatible_hash = u256 { low: reversed_high, high: reversed_low };
+
+            self.solidity_hash_value.write(compatible_hash);
+        }
+
+        fn get_hashed_value(self: @ContractState) -> u256 {
+            self.solidity_hash_value.read()
+        }
```
Comment (julio4) [inline listings/ch02-advanced-concepts/hash_solidity_compatible/src/contract.cairo:39]: `get_hashed_value` and storage `solidity_hash_value` is not necessary, you can remove it and return the hash in the `hash_data` function.

**Review by julio4 — CHANGES_REQUESTED** [review]
LGTM! Thank you very much. I addressed some minimal changes but the overall example is really good.

**Comment by julienbrs** [comment]
Changes requested done!

**Review by julio4 — APPROVED** [review]

**Comment by julio4** [comment]
Thank you for pointing this out! @julienbrs Do you still want to work on this?

**Comment by julienbrs** [comment]
Thanks for the full explanation, I'll start working on it in a few days if it's okay for you @julio4

**Pull request merged**